tiledb/sm/array/filtered_fragment_uris.h

```diff
@@ -0,0 +1,120 @@
+/**
+ * @file   filtered_fragment_uris.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2022 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file defines class FilteredFragmentUris.
+ */
+
+#ifndef TILEDB_FILTERED_FRAGMENT_URIS_H
+#define TILEDB_FILTERED_FRAGMENT_URIS_H
+
+#include "tiledb/common/status.h"
+#include "tiledb/common/thread_pool.h"
+#include "tiledb/sm/filesystem/uri.h"
+#include "tiledb/sm/filesystem/vfs.h"
+
+#include <unordered_map>
+
+using namespace tiledb::common;
+
+namespace tiledb {
+namespace sm {
+
+class ArrayDirectory;
+
+/**
+ * Manages the various URIs inside an array directory, considering
+ * various versions of the on-disk TileDB format.
+ */
+class FilteredFragmentUris {
```

Comment:
Thanks for the fixes! Looks good! One small comment, we could add this class in ArrayDirectory as nested class as it's tightly coupled with array directory internals.